### ISSUE TITLE
IntelliSense is not working

### ISSUE THREAD

**Issue by DenisOstr** [vuejs/language-tools#4338]
IntelliSense doesn't work. Doesn't show what's in the component, nor does it suggest component names in autocomplete (Nuxt UI installed)

![2024-05-03_135051223](https://github.com/vuejs/language-tools/assets/16721761/cb79e45c-7dd6-4816-9d07-15a70d1cde6b)
![2024-05-03_135612245](https://github.com/vuejs/language-tools/assets/16721761/844e818e-5652-4ea1-926c-09cbeaa9e88a)
![2024-05-03_135639474](https://github.com/vuejs/language-tools/assets/16721761/0ea2689a-4cb6-473c-ad3e-52a38c53d9f3)

What I've already tried (Did not help):

- Change extension version (tried all versions)
- Reinstall the extension
- Install a different version of VSCode (1.88)
- Reinstall VSCode
- Disable all extensions and enable only vue - official

Possibly related things:

By TypeScript
![image](https://github.com/vuejs/language-tools/assets/16721761/4c384e0b-e379-405d-85b3-c5e3a6f8b8bc)

By Vue Language Server
![image](https://github.com/vuejs/language-tools/assets/16721761/945300ee-b8e3-4759-8547-0b8969596b7f)

Extension: v2.0.16
VSCode: v1.89.0
OS: Windows 11
Framework: Nuxt 3

**Comment by krxvv** [comment]
Same here after the extension update I'm getting no linting at all, no suggestions/ IntelliSense.

- Extension: v2.0.16
- VSCode: v1.89.0
- OS: Windows 11
- Framework: Quasar

**Comment by edarblanco** [comment]
I have the same problem, just going back to version 2.0.14 solves it, version 2.0.15 is also compromised

**Comment by DenisOstr** [comment]
I tried all the versions that are available on the marketplace. Did not help.

**Comment by krxvv** [comment]
Yo This is so weird I want to say something but first, did you try using any VPN or proxy to another network? 
Like I mean to fix it. try and tell me.

**Comment by DenisOstr** [comment]
This sounds very strange. How changing the network helps the extension, which in theory works offline.
Actually, as expected, VPN didn’t help at all.

**Comment by krxvv** [comment]
Yeah you are right, I know this is nonsense, but for me choosing insider versions also wont work without changing my ip addr.

**Comment by DenisOstr** [comment]
Are you talking about the insider version of the extension or VSCode?
I'm just not really sure what the problem is in the editor or in the extension

**Comment by johnsoncodehk** [comment]
Can you clone https://github.com/johnsoncodehk/volar-starter and reproduce the problem locally?

If yes, can you reproduce this problem for https://github.com/johnsoncodehk/volar-starter in GitHub Codespaces?
If no, can you reproduce this problem for you project in GitHub Codespaces?

**Comment by krxvv** [comment]
@DenisOstr the insider version of this vue extension!

**Comment by DenisOstr** [comment]
> Can you clone https://github.com/johnsoncodehk/volar-starter and reproduce the problem locally?
> 
> If yes, can you reproduce this problem for https://github.com/johnsoncodehk/volar-starter in GitHub Codespaces? If no, can you reproduce this problem for you project in GitHub Codespaces?

I'll try this

**Comment by krxvv** [comment]
> Can you clone https://github.com/johnsoncodehk/volar-starter and reproduce the problem locally?
> 
> If yes, can you reproduce this problem for https://github.com/johnsoncodehk/volar-starter in GitHub Codespaces? If no, can you reproduce this problem for you project in GitHub Codespaces?

Sure, can't right now but I'll also try later

**Comment by arielkern** [comment]
Same issue here. Seems related to vscode april's update (v1.89.0). Downgrading vscode worked for me.

**Comment by DenisOstr** [comment]
I tried and got the following result:

- Locally, it doesn't work either.
- I can’t check it in codespaces, but I tried it in VSCode Web (with only two extensions (Vue - Official and TypeScript IntelliSense for Web)), and it doesn’t work there either

**Comment by DenisOstr** [comment]
> Same issue here. Seems related to vscode april's update (v1.89.0). Downgrading vscode worked for me.

I tried, but I'll try again

**Comment by DenisOstr** [comment]
@arielkern Which version did you install? I tried 1.88.1 but that didn't work either

**Comment by DenisOstr** [comment]
I tried VSCode version 1.88.1, it didn't work.
I tried VSCode version 1.87.2, it didn't work.
I uninstalled VSCode and everything related with it, installed both versions, it didn’t work

**Comment by edarblanco** [comment]
Update VSCode version 1.89.1, it didn't work.
Vue - Official version: v2.0.16

But, VSCode version 1.89.1 and Vue - Official version: v2.0.14 work

**Comment by DenisOstr** [comment]
What exactly works and how?

For me, with versions 1.87.2 and 1.8.27 (VSCode and Vue - Official), only autocomplete worked, and sometimes it doesn’t find certain things. Intellisense also doesn't work for me (when I'm just trying to call something without starting to type. Press CTRL + Space)

On version 1.89.1 I tried all versions of Vue - Official, none worked

**Comment by DenisOstr** [comment]
I've checked it now. It seems like autocomplete works on versions 1.89.1 and 2.0.16, but Intellisense doesn’t show me anything

**Comment by edarblanco** [comment]
I can exactly use autocomplete and so on, without any other changes, but I use Archlinux, I don't know if this could influence
![image](https://github.com/vuejs/language-tools/assets/3819778/31d137cc-2dcc-4c94-8cfe-f1da06e2eeab)

**Comment by edarblanco** [comment]
Hello, a few minutes ago I updated to the latest version of Vue Official v2.0.17 with VS Code 1.89.1 and everything is fine.

**Comment by bosens-China** [comment]
Update to the latest version and set the tsconfig.json file to the following format
```JSON
{
"CompilerOptions":{
"BaseUrl": "./",
"AllowJs": true,
"ResolveJsonModule": true,
//"Target": "ES2022",
"IsolatedModules": true,
"Strict": true,
"Paths":{
"~/*": [./*]
},
"OutDir": "./dist"
}
}
```
It seems that the target will cause an exception to occur

**Comment by fullmooooon** [comment]
delete Vscode plugin "vetur"
use "Vue - official"

**Comment by okmkey45** [comment]
Hello everybody. I was having the same issue and none of the versions worked for me, not even downgrading the extension to 1.8.27

I found that there's a new feature in version 2 called **Hybrid** mode. I found this issue and it's kind of similar (I might be wrong). So, what I did was setting this new setting to `typeScriptPluginOnly`. Probably this is not best fix but at least, it helped me to enable IntelliSense again. So, I hope it can help you guys while this is properly fixed

![Screenshot from 2024-05-30 08-29-50](https://github.com/vuejs/language-tools/assets/22939440/3aa0ae37-be33-480c-8d0f-68dc9ecd87d0)

**Comment by abubakriz** [comment]
I am on VSCode version v1.89.1 and Volar version v2.0.19

It doesn't seem to be able to autocomplete even basic things like auto importing `ref` and other components.

I don't know how I can help beyond saying what's happening to me, is there a way to get a debug log and send it here or something that will help the developers find the source of the issue?

**Comment by dalpii** [comment]
I'm experiencing something related to this, and I would like to provide some findings in hope that makes sense to  maintainers.

I'm using Vue.JS + TypeScript, with default configs (no change regarding Vue/Volar). I'm also using Codespaces for development, so that's why I get this behavior pretty often.

Upon creation of a new Codespaces machine, after everything is installed, if I try to open a Vue file it seems everything is working, hover/IntelliSense on HTML is working fine, but once I try hover on a TypeScript variable I get nothing, and of course, not completions, etc.

Please take noticed on this VSCode status:
<img width="379" alt="Captura de Tela 2024-06-04 às 10 26 38" src="https://github.com/vuejs/language-tools/assets/94379499/fe6876c3-b932-43b1-a5ea-08587621e9cb">

I've enabled language server tracing, and once I hover a simple `const` I get this:
```
[Trace - 1:30:44 PM] Received response 'textDocument/hover - (22)' in 3ms.
No result returned.
```

I've noticed that restarting language server, or event just closing and reopening Codespaces didn't work either.

**But what I found** is that, once I restarted VSCode itself (quit and reopen the application), and load the same Codespaces, and open the same file, everything seems to magically works.

One thing I've noticed is that now I'm getting a new row with TS version on VSCode status:
<img width="392" alt="Captura de Tela 2024-06-04 às 09 59 33" src="https://github.com/vuejs/language-tools/assets/94379499/149048a4-b4dc-481f-aff7-495d3f5a6a4a">

I hope someone can reproduce this even on local codebases, and we get to the bottom of what's going on. Maybe something related to Hybrid Mode?

**Comment by ameryu716** [comment]
When will it be, this problem will resolved? 
This have always helped me, but as far as complementary functions are concerned, really, really, useless.
`ref`,`reactive`,`computed`... various functions exist, but ,I haven't seen any of these codes auto import in the last two years.
Until a few years ago, I was open to using the `vetur` extension as a transition period, but I think that time has passed.
When will this extension work properly?

**Comment by leonward** [comment]
> Hello everybody. I was having the same issue and none of the versions worked for me, not even downgrading the extension to 1.8.27
> 
> I found that there's a new feature in version 2 called **Hybrid** mode. I found this issue and it's kind of similar (I might be wrong). So, what I did was setting this new setting to `typeScriptPluginOnly`. Probably this is not best fix but at least, it helped me to enable IntelliSense again. So, I hope it can help you guys while this is properly fixed
> 
> ![Screenshot from 2024-05-30 08-29-50](https://private-user-images.githubusercontent.com/22939440/335199179-3aa0ae37-be33-480c-8d0f-68dc9ecd87d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzMDI3MDQsIm5iZiI6MTcyMTMwMjQwNCwicGF0aCI6Ii8yMjkzOTQ0MC8zMzUxOTkxNzktM2FhMGFlMzctYmUzMy00ODBjLThkMGYtNjhkYzllY2Q4N2QwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDExMzMyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMjIxNDAxOGNhZmJkOTlkOGQ4NzJkYjU5ZjhhYTQ0MzBkOTJlZWVjNjYwYWJhMzNiYTc0NDVjODk3MjUzNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cxapRnKls2eRhdG09YK1eJclMz6LRi4vndc7Q0vb4a8)

This also allowed me to get suggestions functioning again, unsure of what else it could have broken. It's worth noting that I received a type error when setting this value away from auto, so I've set it to false in my settings.json 
  
` 
 "vue.server.hybridMode": false
`

**Comment by ben1499** [comment]
> > Hello everybody. I was having the same issue and none of the versions worked for me, not even downgrading the extension to 1.8.27
> > I found that there's a new feature in version 2 called **Hybrid** mode. I found this issue and it's kind of similar (I might be wrong). So, what I did was setting this new setting to `typeScriptPluginOnly`. Probably this is not best fix but at least, it helped me to enable IntelliSense again. So, I hope it can help you guys while this is properly fixed
> > ![Screenshot from 2024-05-30 08-29-50](https://private-user-images.githubusercontent.com/22939440/335199179-3aa0ae37-be33-480c-8d0f-68dc9ecd87d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzMDI3MDQsIm5iZiI6MTcyMTMwMjQwNCwicGF0aCI6Ii8yMjkzOTQ0MC8zMzUxOTkxNzktM2FhMGFlMzctYmUzMy00ODBjLThkMGYtNjhkYzllY2Q4N2QwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDExMzMyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMjIxNDAxOGNhZmJkOTlkOGQ4NzJkYjU5ZjhhYTQ0MzBkOTJlZWVjNjYwYWJhMzNiYTc0NDVjODk3MjUzNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cxapRnKls2eRhdG09YK1eJclMz6LRi4vndc7Q0vb4a8)
> 
> This also allowed me to get suggestions functioning again, unsure of what else it could have broken. It's worth noting that I received a type error when setting this value away from auto, so I've set it to false in my settings.json
> 
> ` "vue.server.hybridMode": false`

Setting hybrid mode to false worked for me, but autocomplete for method names (vue 2 options api) doesn't seem to be working.

**Comment by KermanX** [comment]
Closing this issue because there seem to be multiple different problems that have similar behavior. Some of them were months ago and may no longer be applicable.

Possible fixes gathered from comments:

- Upgrade to the latest version
- Check VPN
- Check `compilerOptions.target` field in `tsconfig.json`
- Set `hybrid mode` to `"typeScriptPluginOnly"` or `false` in extension settings - However, this is only a temporary workaround. You may still need to open a new issue with the problem you're facing.

@dalpii Thank you for providing this valuable information. I am not sure if this still happens in the latest version. If it still happens, please open a new issue. Thanks!

**Comment by dalpii** [comment]
I'm still having issues even after checking the possible fixes listed by @KermanX.

But I took some hours today to try to anything else. Although it was hard to replicate with a sample repository, I did these tests on a project we're working on, so quite few files.

My findings are somewhat strange, and I don't even know how to create a bug for this @KermanX, or even if it's a bug here, Codespaces, or something else. So as background, we use Codespaces here with default Microsoft images and configure to install some extensions, including this one.

So, if I create a new Codespace from scratch, wait for it to finish and load things, and try to open a `.vue` file, I get **NO IntelliSense**, no matter what I do (restart Vue server command from VSCode, etc.):

https://github.com/user-attachments/assets/a5251b9f-a8a4-43c4-8c1a-12621fadfac4

But again, if I create a new Codespace from scratch, but this time keep the "Extensions" tab open on VS Code, wait for all installation to finish, making sure to see the "time to load" appear on Vue extension, and then open the same `.vue` file, we can see that now it's showing on status bar that is "initializing JS/TS stuff" and **IntelliSense works right away**:

Sorry, this video is little bit longer:
https://drive.google.com/file/d/1o5fKgwcxu46dbK32Ei7ZhrHhVXurJB8w/view?usp=share_link


I'm really not sure what's going on, but maybe something related to extension initialization?!
I hope these findings help someone with more knowledge on how VS Code extension works.
